tests/unit/model/relationships-test.js

```diff
@@ -66,3 +66,47 @@ test('eachRelatedType() iterates over relations without duplication', function(a
 
   assert.deepEqual(relations, ['occupation', 'person']);
 });
+
+test('normalizing belongsTo relationship names', function(assert) {
+  const UserProfile = DS.Model.extend({
+    user: DS.belongsTo(),
+  });
+
+  let User = DS.Model.extend({
+    userProfile: DS.belongsTo(),
+  });
+
+  store = createStore({
+    user: User,
+    userProfile: UserProfile,
+  });
+
+  User = store.modelFor('user');
+
+  const relationships = get(User, 'relationships');
+  const relationship = relationships.get('user-profile')[0];
+
+  assert.equal(relationship.meta.name, 'userProfile');
+});
+
+test('normalizing hasMany relationship names', function(assert) {
+  const StreamItem = DS.Model.extend({
+    user: DS.belongsTo(),
+  });
+
+  let User = DS.Model.extend({
+    streamItems: DS.hasMany(),
+  });
+
+  store = createStore({
+    user: User,
+    streamItem: StreamItem,
+  });
+
+  User = store.modelFor('user');
+
+  const relationships = get(User, 'relationships');
+  const relationship = relationships.get('stream-item')[0];
```

Comment:
we likely want an assertion that we did in fact normalize to `stream-item` from `streamItems` here, instead of just relying on it implicitly being tested due to how we access. Same with the test above.